napari/utils/transforms/transforms.py

```diff
@@ -597,6 +628,14 @@ def expand_dims(self, axes: Sequence[int]) -> 'Affine':
             name=self.name,
         )
 
+    @cached_property
+    def is_diagonal(self):
```

Comment:
How would you feel about adding an underscore to `is_diagonal`, so that it isn't explicitly part of the public API? At the moment, this property itself is being used as an optimization, and I'm a little unsure if it's something we'll want to support in the long term future if we end up re-designing transforms as in, so I'd lean towards being conservative and not expanding the already large public API here.

I don't feel too strongly about this, as I guess lots of transforms could just return `False` as a default implementation.